modules/core/keeper/keeper.go

```diff
@@ -28,7 +28,7 @@ type Keeper struct {
 
 	cdc codec.BinaryCodec
 
-	ClientKeeper     clientkeeper.Keeper
+	ClientKeeper     *clientkeeper.Keeper
```

Comment:
This change is technically API breaking. The only way to make this PR non API breaking fully would be to put the `consensusHost` implementation on the connection keeper (where's its used for handshake validation) and add a setter at the 03-connection layer. 

However, this might just be more disruptive than this small change in a minor release. 

...still thinking about this.